Sources/RabbitMq/Consumer/RetryingConsumer.swift

```diff
@@ -3,24 +3,28 @@ import AsyncAlgorithms
 import Logging
 import NIOCore
 
-struct RetryingConsumer: Sendable {
+struct RetryingConsumer: ConnectionLoggable {
     let connection: Connection
     let configuration: ConsumerConfiguration
-    let logger: Logger
 
     let retryInterval: Duration
     let consumerWaitChannel = AsyncChannel<Void>()
     let consumeChannel = AsyncChannel<AMQPResponse.Channel.Message.Delivery>()
     let cancellationChannel = AsyncChannel<Void>()
 
+    var logger: Logger {
+        get async {
+            await connection.logger.withMetadata(["queueName": .string(configuration.queueName)])
```

Comment:
This creates a new logger with metadata every time `logger` is used. This is done even when the log is not actually created due to the current log level.